library/init/meta/tactic.lean

```diff
@@ -101,6 +101,11 @@ meta def try_lst : list (tactic unit) → tactic unit
     end
   end
 
+meta def continue (t : tactic α) : tactic (option α) :=
+λ s, result.cases_on (t s)
+ (λ a, success (some a))
+ (λ e ref s', success none s')
+
```

Comment:
@leodemoura Note the `s'` in the last line. This tactic is like `try_core` except that in the case of an error it allows you to continue from the failure state rather than backtracking (hence the name).